airflow/api_connexion/security.py

```diff
@@ -126,13 +126,14 @@ def callback():
             if dag_id or access or access_entity:
                 return access
 
-            # No DAG id is provided, the user is not authorized to access all DAGs and authorization is done
+            # No DAG id is provided: the user is not authorized to access all DAGs and authorization is done
```

Comment:
I don't understand this comment.  Do either of you @jedcunningham @ephraimbuddy ?

Reply:
Yeah I dont understand my own comment myself xD We might want to rephrase it. What I meant is, if no DAG id is provided and user does not have permissions to read all DAGs, we check whether the user have permission to read at least one DAG.

Example: list dags page.

The user does not have permissions to read all DAGs. We want to check if the user has permissions to read at least one DAG. If not, they will have denied access.

To be honest I hate that logic and I was planning to modify it at some point. I think we should introduce a new action `LIST` which gives access to a user to list given resources. That would simplify the logic, if the user has `LIST` permissions on DAGs, then they are authorized to list DAGs. Of course the list of DAGs will be filtered to only contain DAGs that the user has access to (that is already done today).  Today we check whether the user has permission to access one dag to check whether the user has permissions to see the list, and then when listing the DAGs, we check for each DAG, whether they have access to it.
Also, this is a legacy from FAB because we assume here that DAGs are the only type of resources that a user can have resource-specific permission (user X can read on DAG Y). But this is no longer true with the introduction of auth manager where any type of resource can be used to create resource-specific permission. Example: user X can read on connection Z.

Reply:
Right ok so what it means is that we do sort of a partial check.  We check that there's at least one.  But then it's up to the specific endpoint function to implement proper access controls.

> To be honest I hate that logic and I was planning to modify it at some point

Yeah i get it, it seems pretty function for the answer of the check to be `yes` if the user has access to just one entity out of many.  susceptible to the endpoint function not actually doing the verification that is delegated to it.

Reply:
Well there is 2 things I think. The first check checking that the user has access to list these given resources, if not, they get an access denied. Then after, yes the endpoint is responsible of returning only resources that the user has access to. We could remove the first one but as a consequence, the user will get an empty list when trying to lsit something that they do not have access. I am not sure this is a good user experience

Reply:
i think i understand.